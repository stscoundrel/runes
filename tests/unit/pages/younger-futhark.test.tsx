import ReactDOM from 'react-dom'
import YoungerFuthark from 'pages/younger-futhark/index'
import ShortTwig from 'pages/younger-futhark/short-twig'
import LongBranch from 'pages/younger-futhark/long-branch'
import renderer from 'react-test-renderer'

describe('Younger Futhark: default runic keyboard page', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<YoungerFuthark />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<YoungerFuthark />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Younger Futhark: long branch runic keyboard page', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<LongBranch />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<LongBranch />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Younger Futhark: short twig runic keyboard page', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ShortTwig />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<ShortTwig />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
