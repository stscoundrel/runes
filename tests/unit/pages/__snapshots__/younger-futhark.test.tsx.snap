// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Younger Futhark: default runic keyboard page Matches snapshot 1`] = `
Array [
  <main
    className="container"
  >
    <section>
      <header>
        <h1>
          Younger Futhark
        </h1>
        <p>
          Translate runes to text
        </p>
      </header>
      <p
        className="result"
      />
      <div
        className="keyboard"
      >
        <button
          aria-label="Runic key ᚠ, meaning letter f"
          className="key"
          onClick={[Function]}
          title="Runic key ᚠ, meaning letter f"
        >
          ᚠ
        </button>
        <button
          aria-label="Runic key ᚢ, meaning letter u"
          className="key"
          onClick={[Function]}
          title="Runic key ᚢ, meaning letter u"
        >
          ᚢ
        </button>
        <button
          aria-label="Runic key ᚦ, meaning letter þ"
          className="key"
          onClick={[Function]}
          title="Runic key ᚦ, meaning letter þ"
        >
          ᚦ
        </button>
        <button
          aria-label="Runic key ᚬ, meaning letter o"
          className="key"
          onClick={[Function]}
          title="Runic key ᚬ, meaning letter o"
        >
          ᚬ
        </button>
        <button
          aria-label="Runic key ᚱ, meaning letter r"
          className="key"
          onClick={[Function]}
          title="Runic key ᚱ, meaning letter r"
        >
          ᚱ
        </button>
        <button
          aria-label="Runic key ᚴ, meaning letter k"
          className="key"
          onClick={[Function]}
          title="Runic key ᚴ, meaning letter k"
        >
          ᚴ
        </button>
        <button
          aria-label="Runic key ᚼ, meaning letter h"
          className="key"
          onClick={[Function]}
          title="Runic key ᚼ, meaning letter h"
        >
          ᚼ
        </button>
        <button
          aria-label="Runic key ᚽ, meaning letter h"
          className="key"
          onClick={[Function]}
          title="Runic key ᚽ, meaning letter h"
        >
          ᚽ
        </button>
        <button
          aria-label="Runic key ᚾ, meaning letter n"
          className="key"
          onClick={[Function]}
          title="Runic key ᚾ, meaning letter n"
        >
          ᚾ
        </button>
        <button
          aria-label="Runic key ᚿ, meaning letter n"
          className="key"
          onClick={[Function]}
          title="Runic key ᚿ, meaning letter n"
        >
          ᚿ
        </button>
        <button
          aria-label="Runic key ᛁ, meaning letter i"
          className="key"
          onClick={[Function]}
          title="Runic key ᛁ, meaning letter i"
        >
          ᛁ
        </button>
        <button
          aria-label="Runic key ᛅ, meaning letter a"
          className="key"
          onClick={[Function]}
          title="Runic key ᛅ, meaning letter a"
        >
          ᛅ
        </button>
        <button
          aria-label="Runic key ᛆ, meaning letter a"
          className="key"
          onClick={[Function]}
          title="Runic key ᛆ, meaning letter a"
        >
          ᛆ
        </button>
        <button
          aria-label="Runic key ᛋ, meaning letter s"
          className="key"
          onClick={[Function]}
          title="Runic key ᛋ, meaning letter s"
        >
          ᛋ
        </button>
        <button
          aria-label="Runic key ᛌ, meaning letter s"
          className="key"
          onClick={[Function]}
          title="Runic key ᛌ, meaning letter s"
        >
          ᛌ
        </button>
        <button
          aria-label="Runic key ᛏ, meaning letter t"
          className="key"
          onClick={[Function]}
          title="Runic key ᛏ, meaning letter t"
        >
          ᛏ
        </button>
        <button
          aria-label="Runic key ᛐ, meaning letter t"
          className="key"
          onClick={[Function]}
          title="Runic key ᛐ, meaning letter t"
        >
          ᛐ
        </button>
        <button
          aria-label="Runic key ᛒ, meaning letter b"
          className="key"
          onClick={[Function]}
          title="Runic key ᛒ, meaning letter b"
        >
          ᛒ
        </button>
        <button
          aria-label="Runic key ᛘ, meaning letter m"
          className="key"
          onClick={[Function]}
          title="Runic key ᛘ, meaning letter m"
        >
          ᛘ
        </button>
        <button
          aria-label="Runic key ᛚ, meaning letter l"
          className="key"
          onClick={[Function]}
          title="Runic key ᛚ, meaning letter l"
        >
          ᛚ
        </button>
        <button
          aria-label="Runic key ᛦ, meaning letter R"
          className="key"
          onClick={[Function]}
          title="Runic key ᛦ, meaning letter R"
        >
          ᛦ
        </button>
        <button
          aria-label="Runic key :, meaning letter  "
          className="key"
          onClick={[Function]}
          title="Runic key :, meaning letter  "
        >
          :
        </button>
      </div>
    </section>
  </main>,
  <footer
    className="section"
  >
    <div
      className="container text-center"
    >
      <small
        className="copyright"
      >
        Copyright © 
        2021
        <a
          href="https://github.com/stscoundrel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sampo Silvennoinen / StScoundrel
        </a>
      </small>
    </div>
  </footer>,
]
`;

exports[`Younger Futhark: long branch runic keyboard page Matches snapshot 1`] = `
Array [
  <main
    className="container"
  >
    <section>
      <header>
        <h1>
          Younger Futhark - Long Branch
        </h1>
        <p>
          Translate runes to text
        </p>
      </header>
      <p
        className="result"
      />
      <div
        className="keyboard"
      >
        <button
          aria-label="Runic key ᚠ, meaning letter f"
          className="key"
          onClick={[Function]}
          title="Runic key ᚠ, meaning letter f"
        >
          ᚠ
        </button>
        <button
          aria-label="Runic key ᚢ, meaning letter u"
          className="key"
          onClick={[Function]}
          title="Runic key ᚢ, meaning letter u"
        >
          ᚢ
        </button>
        <button
          aria-label="Runic key ᚦ, meaning letter þ"
          className="key"
          onClick={[Function]}
          title="Runic key ᚦ, meaning letter þ"
        >
          ᚦ
        </button>
        <button
          aria-label="Runic key ᚬ, meaning letter o"
          className="key"
          onClick={[Function]}
          title="Runic key ᚬ, meaning letter o"
        >
          ᚬ
        </button>
        <button
          aria-label="Runic key ᚱ, meaning letter r"
          className="key"
          onClick={[Function]}
          title="Runic key ᚱ, meaning letter r"
        >
          ᚱ
        </button>
        <button
          aria-label="Runic key ᚴ, meaning letter k"
          className="key"
          onClick={[Function]}
          title="Runic key ᚴ, meaning letter k"
        >
          ᚴ
        </button>
        <button
          aria-label="Runic key ᚼ, meaning letter h"
          className="key"
          onClick={[Function]}
          title="Runic key ᚼ, meaning letter h"
        >
          ᚼ
        </button>
        <button
          aria-label="Runic key ᚾ, meaning letter n"
          className="key"
          onClick={[Function]}
          title="Runic key ᚾ, meaning letter n"
        >
          ᚾ
        </button>
        <button
          aria-label="Runic key ᛁ, meaning letter i"
          className="key"
          onClick={[Function]}
          title="Runic key ᛁ, meaning letter i"
        >
          ᛁ
        </button>
        <button
          aria-label="Runic key ᛅ, meaning letter a"
          className="key"
          onClick={[Function]}
          title="Runic key ᛅ, meaning letter a"
        >
          ᛅ
        </button>
        <button
          aria-label="Runic key ᛋ, meaning letter s"
          className="key"
          onClick={[Function]}
          title="Runic key ᛋ, meaning letter s"
        >
          ᛋ
        </button>
        <button
          aria-label="Runic key ᛏ, meaning letter t"
          className="key"
          onClick={[Function]}
          title="Runic key ᛏ, meaning letter t"
        >
          ᛏ
        </button>
        <button
          aria-label="Runic key ᛒ, meaning letter b"
          className="key"
          onClick={[Function]}
          title="Runic key ᛒ, meaning letter b"
        >
          ᛒ
        </button>
        <button
          aria-label="Runic key ᛘ, meaning letter m"
          className="key"
          onClick={[Function]}
          title="Runic key ᛘ, meaning letter m"
        >
          ᛘ
        </button>
        <button
          aria-label="Runic key ᛚ, meaning letter l"
          className="key"
          onClick={[Function]}
          title="Runic key ᛚ, meaning letter l"
        >
          ᛚ
        </button>
        <button
          aria-label="Runic key ᛦ, meaning letter R"
          className="key"
          onClick={[Function]}
          title="Runic key ᛦ, meaning letter R"
        >
          ᛦ
        </button>
        <button
          aria-label="Runic key :, meaning letter  "
          className="key"
          onClick={[Function]}
          title="Runic key :, meaning letter  "
        >
          :
        </button>
      </div>
    </section>
  </main>,
  <footer
    className="section"
  >
    <div
      className="container text-center"
    >
      <small
        className="copyright"
      >
        Copyright © 
        2021
        <a
          href="https://github.com/stscoundrel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sampo Silvennoinen / StScoundrel
        </a>
      </small>
    </div>
  </footer>,
]
`;

exports[`Younger Futhark: short twig runic keyboard page Matches snapshot 1`] = `
Array [
  <main
    className="container"
  >
    <section>
      <header>
        <h1>
          Younger Futhark - Short Twig
        </h1>
        <p>
          Translate runes to text
        </p>
      </header>
      <p
        className="result"
      />
      <div
        className="keyboard"
      >
        <button
          aria-label="Runic key ᚠ, meaning letter f"
          className="key"
          onClick={[Function]}
          title="Runic key ᚠ, meaning letter f"
        >
          ᚠ
        </button>
        <button
          aria-label="Runic key ᚢ, meaning letter u"
          className="key"
          onClick={[Function]}
          title="Runic key ᚢ, meaning letter u"
        >
          ᚢ
        </button>
        <button
          aria-label="Runic key ᚦ, meaning letter þ"
          className="key"
          onClick={[Function]}
          title="Runic key ᚦ, meaning letter þ"
        >
          ᚦ
        </button>
        <button
          aria-label="Runic key ᚬ, meaning letter o"
          className="key"
          onClick={[Function]}
          title="Runic key ᚬ, meaning letter o"
        >
          ᚬ
        </button>
        <button
          aria-label="Runic key ᚱ, meaning letter r"
          className="key"
          onClick={[Function]}
          title="Runic key ᚱ, meaning letter r"
        >
          ᚱ
        </button>
        <button
          aria-label="Runic key ᚴ, meaning letter k"
          className="key"
          onClick={[Function]}
          title="Runic key ᚴ, meaning letter k"
        >
          ᚴ
        </button>
        <button
          aria-label="Runic key ᚽ, meaning letter h"
          className="key"
          onClick={[Function]}
          title="Runic key ᚽ, meaning letter h"
        >
          ᚽ
        </button>
        <button
          aria-label="Runic key ᚿ, meaning letter n"
          className="key"
          onClick={[Function]}
          title="Runic key ᚿ, meaning letter n"
        >
          ᚿ
        </button>
        <button
          aria-label="Runic key ᛁ, meaning letter i"
          className="key"
          onClick={[Function]}
          title="Runic key ᛁ, meaning letter i"
        >
          ᛁ
        </button>
        <button
          aria-label="Runic key ᛆ, meaning letter a"
          className="key"
          onClick={[Function]}
          title="Runic key ᛆ, meaning letter a"
        >
          ᛆ
        </button>
        <button
          aria-label="Runic key ᛌ, meaning letter s"
          className="key"
          onClick={[Function]}
          title="Runic key ᛌ, meaning letter s"
        >
          ᛌ
        </button>
        <button
          aria-label="Runic key ᛏ, meaning letter t"
          className="key"
          onClick={[Function]}
          title="Runic key ᛏ, meaning letter t"
        >
          ᛏ
        </button>
        <button
          aria-label="Runic key ᛐ, meaning letter t"
          className="key"
          onClick={[Function]}
          title="Runic key ᛐ, meaning letter t"
        >
          ᛐ
        </button>
        <button
          aria-label="Runic key ᛒ, meaning letter b"
          className="key"
          onClick={[Function]}
          title="Runic key ᛒ, meaning letter b"
        >
          ᛒ
        </button>
        <button
          aria-label="Runic key ᛘ, meaning letter m"
          className="key"
          onClick={[Function]}
          title="Runic key ᛘ, meaning letter m"
        >
          ᛘ
        </button>
        <button
          aria-label="Runic key ᛚ, meaning letter l"
          className="key"
          onClick={[Function]}
          title="Runic key ᛚ, meaning letter l"
        >
          ᛚ
        </button>
        <button
          aria-label="Runic key ᛦ, meaning letter R"
          className="key"
          onClick={[Function]}
          title="Runic key ᛦ, meaning letter R"
        >
          ᛦ
        </button>
        <button
          aria-label="Runic key :, meaning letter  "
          className="key"
          onClick={[Function]}
          title="Runic key :, meaning letter  "
        >
          :
        </button>
      </div>
    </section>
  </main>,
  <footer
    className="section"
  >
    <div
      className="container text-center"
    >
      <small
        className="copyright"
      >
        Copyright © 
        2021
        <a
          href="https://github.com/stscoundrel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sampo Silvennoinen / StScoundrel
        </a>
      </small>
    </div>
  </footer>,
]
`;
